static/app/components/replays/videoReplayer.tsx

```diff
@@ -288,6 +288,11 @@ export class VideoReplayer {
     // Show the next video
     this.showVideo(nextVideo);
 
+    // Preload the next few videos
+    if (index < this._attachments.length) {
+      this.getOrCreateVideo(index + 1);
```

Comment:
🤔 when a video ends, we call `handleSegmentEnd`, which calls `playSegmentAtIndex` -> `getOrCreateVideo`, so shouldn't the next set of videos get preloaded?

Reply:
i think maybe it doesn't do it far ahead enough - looks like it's only saying "get the next video" at the exact moment when we need it, but that leads to some buffering. what i wanted to do was preload early so we avoided any of that loading by having things ready before the segment even ends - does that make sense?

Reply:
yeah what you're saying (+ the code here) makes sense, I'm trying to understand why the existing code doesn't work as expected. When we play, we should have 0, 1, 2, preloaded, when segment 0 ends, segment 1 plays, so we should be preloading 2, 3, 4 right?

Reply:
i believe it's because in our getOrCreateVideo function, if we see a video is available in our dictionary we'll just return it and that's it -- in the case of video 1, we see we've loaded it already and so just return it. we don't call the rest of the function which does the preloading of future videos

Reply:
ohhh right, so this would wait until segment 2 before we would preload segment 3. What do you think about removing the preloading side-effect from `getOrCreateVideo` and explicitly calling `createVideoForRange` as needed? The three situations we've identified right now where we need to preload:

* Immediately when player is initialized with attachments so that it is ready to play
* When seeking we want to preload in the forward direction first as this will be the most common action
  * we also want to preload behind in case they want to rewind
* When playing the next segment, we should preload and maintain the PRELOAD_BUFFER size in the forward direction only

(maybe it makes sense to rename createVideoForRange to preloadVideos or something with preload?)

Reply:
yeah makes sense to me!

Reply:
updated, although it does make the code less clean

Reply:
It's more code, but I think the explicitness might be good